### ISSUE TITLE
Support shadow plugin

### ISSUE THREAD

**Issue by msangel** [neworld/gradle-jdeploy-plugin#4]
This plugin gets the target executable jar from the project build and that contains only compiled code of current project BUT not its dependencies. So target command also would be simple jar without its dependencies. 
But If my project use some libraries and I want to bundle them with target JAR(there are a lot of option for that) the [shadow plugin](https://github.com/johnrengelman/shadow) is used most commonly.
And if that plugin is present:
1) the `./gradlew jdeployInstall` command must depend on `./gradlew build`. The shadow plugin applies himself as a part of the build phase. Well,  this can be made manually with adding `tasks.jdeployInstall.dependsOn(build)` to build.gradle. But truly says, I expected this behavior as default. 
2) `jar` option of this plugin must be taken from default or configured value of shadow's "Output Name" option. The default option is as simple as file("$buildDir/libs/$name-$version-all.jar"):
```groovy
jdeploy {
    // ...
    jar = file("$buildDir/libs/$name-$version-all.jar")
} 
```
But this can be [configured](https://imperceptiblethoughts.com/shadow/configuration/#configuring-output-name) and the plugin must get real executable properly.

**Comment by msangel** [comment]
Not sure if this is an issue, as in the end, I solve all problems by simply added 2 lines of config, BUT I spent time on: figure out this plugin options,  applying them properly, reading documentation/source code of this and the `shadow` plugin. 
And the behavior I expected was intuitive so I believe these options must be added to this project as default ones if the `shadow` plugin is present.

**Comment by neworld** [comment]
Thanks for the feedback. I definitely include instructions on how to use the custom jar.

But you are right; this plugin intended to work automatically as much as possible. However, I am not familiar with the shadow plugin. I am always [collecting all libs using `jar` config](https://github.com/neworld/vd2svg/blob/master/build.gradle#L38-L40) 
```kotlin
jar {
    from {
        configurations.compile.collect { it.isDirectory() ? it : zipTree(it) }
    }
}
```

Maybe do you have an idea of how this plugin should play with shadow plugin nicely? Or maybe, do you have open source project and could show your config? It could be beneficial.

**Comment by msangel** [comment]
I have a project with both this and shadow plugin: https://github.com/msangel/javyll/blob/master/build.gradle
The shadow plugin is straightforward: you just apply it and it creates an additional jar with ` -all` name suffix (default). It also smart. If you have the "application" plugin, it read all the settings from there and apply to his final jar. (edited: it still build two versions of jar)

**Comment by msangel** [comment]
Well, I think that default expected behavior for this plugin when the shadow is present is to install his artifact (with -all suffix) instead of those without dependencies.

**Comment by neworld** [comment]
I am thinking about the same behavior. This plugin should depend on `shadowJar` task and pick up jar location from `shadowJar`

**Comment by neworld** [comment]
Add support of Shadow plugin, which takes priority over jar task. Released with 0.6.0.
